### PR TITLE
Update README.md to add info about dynamic allocation on YARN

### PR DESCRIPTION
**Pull request by akiezun**
added info about dynamic allocation on YARN (I tested it on Google and it works nicely)

@lbergelson please review

### CODE REVIEW

**Comment by lbergelson** [comment]
neat

File: README.md [hunk 1]
```diff
@@ -111,6 +111,7 @@ If you are looking for the codebase of the current production version of GATK, p
             --conf spark.yarn.executor.memoryOverhead=600
         ```
 
+    * You can also omit the "--num-executors" to enable [dynamic allocation](https://spark.apache.org/docs/latest/job-scheduling.html#dynamic-resource-allocation) if you configure the cluster properly (see the Spark website for instructins).
```
Comment (lbergelson) [inline README.md]: Can you also omit memory and core arguments then?

Comment (akiezun) [inline README.md]: I think not. @tomwhite do you know if you can meaningfully omit  memory and core args with dynamic allocation?

Comment (tomwhite) [inline README.md]: I don't think so. The number of executors is the the thing that's dynamic, not the size of an executor.

**Pull request merged**